docker-entrypoint.sh

```diff
@@ -0,0 +1,447 @@
+        }
+    fi
+}
+
+# Validate API configuration if enabled
+validate_api_config() {
+    local api_enabled=$(echo "${PA_API_ENABLED:-false}" | tr '[:upper:]' '[:lower:]')
+    if [ "$api_enabled" = "true" ] && [ -n "${PA_PDNS_API_URL}" ]; then
+        [ -z "${PA_PDNS_API_KEY}" ] && {
+            log "ERROR: PA_PDNS_API_KEY is required when PowerDNS API URL is specified"
+            exit 1
+        }
+    fi
+}
+
+# Validate LDAP configuration if enabled
+validate_ldap_config() {
+    local ldap_enabled=$(echo "${PA_LDAP_ENABLED:-false}" | tr '[:upper:]' '[:lower:]')
+    if [ "$ldap_enabled" = "true" ]; then
+        local required_ldap_vars=("PA_LDAP_URI" "PA_LDAP_BASE_DN")
+        for var in "${required_ldap_vars[@]}"; do
+            [ -z "${!var}" ] && {
+                log "ERROR: ${var} is required when LDAP is enabled"
+                exit 1
+            }
+        done
+    fi
+}
+
+# Create initial admin user if specified
+create_admin_user() {
+    local create_admin=$(echo "${PA_CREATE_ADMIN:-false}" | tr '[:upper:]' '[:lower:]')
+
+    if [ "$create_admin" != "true" ] && [ "$create_admin" != "1" ] && [ "$create_admin" != "yes" ]; then
+        debug_log "Admin user creation disabled"
+        return 0
+    fi
+
+    local admin_username="${PA_ADMIN_USERNAME:-admin}"
+    local admin_password="${PA_ADMIN_PASSWORD:-admin}"
+    local admin_email="${PA_ADMIN_EMAIL:-admin@example.com}"
+    local admin_fullname="${PA_ADMIN_FULLNAME:-Administrator}"
+
+    debug_log "Creating admin user: ${admin_username}"
+
+    # Generate password hash using PHP
+    local password_hash
+    local temp_php_file="/tmp/hash_password.php"
+    echo "<?php echo password_hash('${admin_password}', PASSWORD_DEFAULT);" > "${temp_php_file}"
+    password_hash=$(php -d error_reporting=0 -d display_startup_errors=0 "${temp_php_file}" 2>/dev/null)
+    rm -f "${temp_php_file}"
+
+    if [ $? -ne 0 ] || [ -z "${password_hash}" ]; then
+        log "ERROR: Failed to generate password hash for admin user"
+        exit 1
+    fi
+
+    debug_log "Generated password hash for admin user"
+
+    # Database-specific user creation
+    case "${DB_TYPE:-sqlite}" in
+        "sqlite")
+            local db_file="${DB_FILE:-/db/pdns.db}"
+            debug_log "Creating admin user in SQLite database: ${db_file}"
+
+            # Check if user already exists
+            local user_exists
+            user_exists=$(sqlite3 "${db_file}" "SELECT COUNT(*) FROM users WHERE username='${admin_username}';")
+
+            if [ "${user_exists}" -gt 0 ]; then
+                log "Admin user '${admin_username}' already exists, skipping creation"
+                return 0
+            fi
+
+            # Insert admin user
+            sqlite3 "${db_file}" "INSERT INTO users (username, password, fullname, email, description, perm_templ, active, use_ldap) VALUES ('${admin_username}', '${password_hash}', '${admin_fullname}', '${admin_email}', 'System Administrator', 1, 1, 0);"
```

Comment:
Do you think it's still safe to allow direct values without escaping? I'm not sure if prepared statements are an option here, but maybe.